src/hotspot/share/oops/symbol.cpp

```diff
@@ -390,11 +389,10 @@ void Symbol::print() const { print_on(tty); }
 // The print_value functions are present in all builds, to support the
 // disassembler and error reporting.
 void Symbol::print_value_on(outputStream* st) const {
-  st->print("'");
-  for (int i = 0; i < utf8_length(); i++) {
-    st->print("%c", char_at(i));
-  }
-  st->print("'");
+  st->print_raw("'", 1);
+  static_assert(sizeof(u1) == sizeof(char), "must be");
```

Comment:
Given the whole class assumes u1 and char equivalence this assertion seems out of place here.

Reply:
I removed the static assert.